### DISCUSSION TITLE
Provision in advance/Pre heat

### DISCUSSION THREAD

**Discussion by yatanasov-hs** [kedacore/keda#4852] [Q&A / Need Help] [answered]
I am using Keda 2.11 for scaling pods based on messages in AWS SQS.  My Scaled obj definition looks like:

`apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: sample
  namespace: sample-namespace
spec:
  maxReplicaCount: 10
  minReplicaCount: 0
  scaleTargetRef:
    name: sample
  triggers:
  - metadata:
      awsRegion: sample_region
      identityOwner: operator
      queueLength: "1"
      queueURL: sample_queue
    type: aws-sqs-queue
`

Is there a way to always scale at calculated value + Constant. For example when there are no messages it will scale at Constant. When there are 2 messages it will scale to 2 + Constant. That would mean we will be always over provisioned by Constant pods. This will help me not have any delay when processing messages.

**Comment by JorTurFer** [comment]
Hello,
Does your workload process one message in parallel? I ask because I see `queueLength` and that means a pod per message. If this is your case, ScaledJobs currently support that feature. Using ScaledJob, you could set minReplicaCount:2 and you will have always 2 empty jobs, ready for processing the incoming mesages.
If you need to use ScaledObject for any reason, currently there isn't any built-in feature. [This feature could help you in the near future](https://github.com/kedacore/keda/pull/4821), but it's not available yet.
If you are in rush, you could use [Metrics API scaler](https://keda.sh/docs/2.11/scalers/metrics-api/) to wrap SQS API with your own API, modifying the value with your custom logic. I mean, your API could just query SQS API and add 2 to the result.

**Comment by yatanasov-hs** [comment, marked as answer]
Correct, my worker processes 1 message at a time.
I'd rather not use jobs since there is caching in deployments which I would lose when running a job per message. Hoping to save time implementing custom api server.
